docs/sidecar/comparison.md

```diff
@@ -0,0 +1,407 @@
+        'Authorization': `Bearer ${incomingToken}`
+      }
+    }
+  );
+  
+  const result = await response.json();
+  return JSON.parse(result.content);
+}
+```
+
+## Feature Comparison
+
+| Feature | Microsoft.Identity.Web | Sidecar |
+|---------|------------------------|---------|
+| **Language Support** | C# / .NET only | Any language (HTTP) |
+| **Deployment** | In-process library | Separate container |
+| **Token Acquisition** | ✅ Direct MSAL.NET | ✅ Via HTTP API |
+| **Token Caching** | ✅ In-memory, distributed | ✅ In-memory, distributed |
+| **OBO Flow** | ✅ Native support | ✅ Via HTTP endpoint |
+| **Client Credentials** | ✅ Native support | ✅ Via HTTP endpoint |
+| **Managed Identity** | ✅ Direct support | ✅ Direct support |
+| **Agent Identities** | ✅ Via extensions | ✅ Query parameters |
+| **Token Validation** | ✅ Middleware | ✅ /Validate endpoint |
+| **Downstream API** | ✅ IDownstreamApi | ✅ /DownstreamApi endpoint |
+| **Microsoft Graph** | ✅ Graph SDK integration | ⚠️ Via DownstreamApi |
+| **Performance** | ⚡ In-process (fastest) | 🔄 HTTP overhead |
+| **Configuration** | appsettings.json, code | appsettings.json, Environment variables |
+| **Debugging** | ✅ Standard .NET debugging | ⚠️ Container debugging |
+| **Hot Reload** | ✅ .NET Hot Reload | ❌ Container restart |
+| **Package Updates** | 📦 NuGet packages | 🐳 Container images |
+| **License** | MIT | MIT |
+
+## When to Use Each Approach
+
+### Use Microsoft.Identity.Web (In-Process) When:
+
+1. **Building .NET Applications Exclusively**
+   - ASP.NET Core Web APIs
+   - ASP.NET Core Web Apps
+   - .NET Worker Services
+   - Blazor applications
+   - Daemon apps
+
+2. **Performance is Critical**
+   - High-throughput scenarios
+   - Latency-sensitive operations
+   - Every millisecond counts
+
+3. **Deep Integration Required**
+   - Custom MSAL.NET configuration
+   - Direct access to MSAL features
+   - Advanced token cache strategies
+
+4. **Rapid Development**
+   - Quick prototyping
+   - Hot reload for development
+   - Standard .NET debugging
+
+5. **Single-Language Stack**
+   - Team expertise in C#/.NET
+   - No multi-language requirements
+
+### Use Sidecar (Out-of-Process) When:
+
+1. **Multi-Language Microservices**
+   - Node.js, Python, Go, Java services
+   - Polyglot architectures
+   - Services in multiple languages
+
+2. **Containerized Deployments**
+   - Kubernetes environments
+   - Docker Compose setups
+   - Service mesh architectures
+
+3. **Centralized Authentication Logic**
+   - Standardized auth across services
+   - Consistent token handling
+   - Shared configuration management
+
+4. **Technology Diversity**
+   - Mix of frameworks and languages
+   - Non-.NET services
+   - Legacy systems integration
+
+5. **Operational Benefits**
+   - Independent scaling of auth logic
+   - Separate auth updates from app code
+   - Centralized monitoring of auth
+
+## Migration Guidance
+
+### Migrating from Microsoft.Identity.Web to Sidecar
+
```

Comment:
Microsoft doesn't recommend that you move from Microsoft.Identity.Web to the sidecar, but if you chose to do so, here is how you would do. This also gives an idea on how you would do in other languages and framework as the concepts are similar.